lib/service/database.js

```diff
@@ -321,7 +325,8 @@ DatabaseService.prototype.queueAnalysisOperations = function(analysis, callback)
                     query: annotateNodeType(nodeToUpdate) + transactionQuery([
                         deleteFromCacheTableQuery(targetTableName),
                         populateCacheTableQuery(targetTableName, nodeToUpdate.sql()),
-                        checkCacheTableQuery(targetTableName)
+                        checkCacheTableQuery(targetTableName),
+                        analyzeTableQuery(targetTableName)
```

Comment:
That's a good one. Let me think about the **possible failure modes**:
- **timeout?** I'd say the whole thing should fail. It'll take a small fraction of the analysis time (in [benchmarks](https://gist.github.com/rafatower/efbaa2924b3a1cc7b7ea0698adafe3b9), on a table of 100M rows it took 22.4 seconds). Of course there's a "corner case" in which the analysis is right below the timeout and the `ANALYZE` would make cross the timeout threshold, but...
- **permissions?** if the role has perms to write the cache table, it shall also have perms to run `ANALYZE`
- **??? any other thing?** (disk full, failover, OOM, corruptions, etc.) IMO it is best not to handle those. It is debatable but there are reasons not to do so (i.e: let the exception bubble and let the RDBMS do it's recovery stuff).

So, my vote goes for not adding exception handling code here. We can always add it later on. What's your view here?